pandas/tests/frame/test_query_eval.py

```diff
@@ -160,144 +165,175 @@ def test_eval_resolvers_as_list(self):
-        b = np.random.choice(['eggs', 'ham'], size=10)
+        a = np.random.choice(['red', 'a_red', 'a_red_a',
+                              'red_a', 'Red', 'green'], size=30)
+        b = np.random.choice(['eggs', 'ham'], size=30)
         index = MultiIndex.from_arrays([a, b], names=['color', 'food'])
-        df = DataFrame(np.random.randn(10, 2), index=index)
+        df = DataFrame(np.random.randn(30, 2), index=index)
         ind = Series(df.index.get_level_values('color').values, index=index,
                      name='color')
+        kwargs = {'parser': parser, 'engine': engine,
+                  'partial_str_match': partial_str_match}
 
         # equality
-        res1 = df.query('color == "red"', parser=parser, engine=engine)
-        res2 = df.query('"red" == color', parser=parser, engine=engine)
+        res1 = df.query('color == "red"', **kwargs)
+        res2 = df.query('"red" == color', **kwargs)
         exp = df[ind == 'red']
         assert_frame_equal(res1, exp)
         assert_frame_equal(res2, exp)
 
         # inequality
-        res1 = df.query('color != "red"', parser=parser, engine=engine)
-        res2 = df.query('"red" != color', parser=parser, engine=engine)
+        res1 = df.query('color != "red"', **kwargs)
+        res2 = df.query('"red" != color', **kwargs)
         exp = df[ind != 'red']
         assert_frame_equal(res1, exp)
         assert_frame_equal(res2, exp)
 
         # list equality (really just set membership)
-        res1 = df.query('color == ["red"]', parser=parser, engine=engine)
-        res2 = df.query('["red"] == color', parser=parser, engine=engine)
+        res1 = df.query('color == ["red"]', **kwargs)
+        res2 = df.query('["red"] == color', **kwargs)
         exp = df[ind.isin(['red'])]
         assert_frame_equal(res1, exp)
         assert_frame_equal(res2, exp)
 
-        res1 = df.query('color != ["red"]', parser=parser, engine=engine)
-        res2 = df.query('["red"] != color', parser=parser, engine=engine)
+        res1 = df.query('color != ["red"]', **kwargs)
+        res2 = df.query('["red"] != color', **kwargs)
         exp = df[~ind.isin(['red'])]
         assert_frame_equal(res1, exp)
         assert_frame_equal(res2, exp)
 
         # in/not in ops
-        res1 = df.query('["red"] in color', parser=parser, engine=engine)
-        res2 = df.query('"red" in color', parser=parser, engine=engine)
-        exp = df[ind.isin(['red'])]
-        assert_frame_equal(res1, exp)
-        assert_frame_equal(res2, exp)
-
-        res1 = df.query('["red"] not in color', parser=parser, engine=engine)
-        res2 = df.query('"red" not in color', parser=parser, engine=engine)
-        exp = df[~ind.isin(['red'])]
-        assert_frame_equal(res1, exp)
-        assert_frame_equal(res2, exp)
+        res1 = df.query('["red"] in color', **kwargs)
+        res2 = df.query('"red" in color', **kwargs)
+        exp1 = df[ind.isin(['red'])]
+        if partial_str_match:
```

Comment:
This test is pretty huge already. I think it would be better to create a separate test for this